tests/Sulu/Component/Rest/ListBuilder/Metadata/Doctrine/Driver/XmlDriverTest.php

```diff
@@ -0,0 +1,193 @@
+        self::assertNull($join->getCondition());
+        self::assertEquals('WITH', $join->getConditionMethod());
+        self::assertEquals('LEFT', $join->getMethod());
+
+        self::assertInstanceOf(ConcatenationType::class, $result->propertyMetadata['fullName']->getType());
+        self::assertEquals(' ', $result->propertyMetadata['fullName']->getType()->getGlue());
+        self::assertCount(2, $result->propertyMetadata['fullName']->getType()->getFields());
+
+        $field = $result->propertyMetadata['fullName']->getType()->getFields()[0];
+        self::assertEquals('firstName', $field->getName());
+        self::assertEquals(
+            'SuluContactBundle:Contact',
+            $field->getEntityName()
+        );
+        self::assertEmpty($field->getJoins());
+
+        $field = $result->propertyMetadata['fullName']->getType()->getFields()[1];
+        self::assertEquals('lastName', $field->getName());
+        self::assertEquals(
+            'SuluContactBundle:Contact',
+            $field->getEntityName()
+        );
+        self::assertEmpty($field->getJoins());
+
+        self::assertEquals('city', $result->propertyMetadata['city']->getType()->getField()->getName());
+        self::assertEquals(
+            'SuluContactBundle:Address',
+            $result->propertyMetadata['city']->getType()->getField()->getEntityName()
+        );
+        self::assertCount(2, $result->propertyMetadata['city']->getType()->getField()->getJoins());
+
+        $join = $result->propertyMetadata['city']->getType()->getField()->getJoins()[0];
+        self::assertEquals('SuluContactBundle:ContactAddress', $join->getEntityName());
+        self::assertEquals('SuluContactBundle:Contact.contactAddresses', $join->getEntityField());
+        self::assertEquals('SuluContactBundle:ContactAddress.main = TRUE', $join->getCondition());
+        self::assertEquals('WITH', $join->getConditionMethod());
+        self::assertEquals('LEFT', $join->getMethod());
+
+        $join = $result->propertyMetadata['city']->getType()->getField()->getJoins()[1];
+        self::assertEquals('SuluContactBundle:Address', $join->getEntityName());
+        self::assertEquals('SuluContactBundle:ContactAddress.address', $join->getEntityField());
+        self::assertNull($join->getCondition());
+        self::assertEquals('WITH', $join->getConditionMethod());
+        self::assertEquals('LEFT', $join->getMethod());
+    }
+
+    public function testLoadMetadataFromFileEmpty()
+    {
+        $locator = $this->prophesize(FileLocatorInterface::class);
+        $parameterBag = $this->prophesize(ParameterBagInterface::class);
+
+        $parameterBag->resolveValue(Argument::any())->willReturnArgument(0);
+
+        $driver = new XmlDriver($locator->reveal(), $parameterBag->reveal());
+
+        $reflectionMethod = new \ReflectionMethod(get_class($driver), 'loadMetadataFromFile');
+        $reflectionMethod->setAccessible(true);
```

Comment:
I would prefer never to use Reflection to access protected / private methods, tests should not know about the internals of classes. The following would be cleaner I think:


```php
public function setUp()
{
    $filePath = '/foo/bar/bar';
    $this->reflection = $this->prophesize('ReflectionClass');

    $locator = $this->prophesize('...\Locator');
    $this->locator->findFileForClass($reflection->reveal(), 'xml')->willReturn($filePath);

    $this->driver = new XmlDriver($locator->reveal());
}

public function testSomething()
{
    $this->driver>loadMetadataForClass($this->reflection->reveal());
    // ...
}
```